### DISCUSSION TITLE
configuration documentation

### DISCUSSION THREAD

**Discussion by rwv37** [smallstep/certificates#1862] [Q&A] [answered]
Is there any documentation on how to configure the server more detailed and/or complete than what's on the "[Configuring step-ca](https://smallstep.com/docs/step-ca/configuration/)" page?

That page is helpful, but it's vague and incomplete in a lot of ways. For example (and just an example), the entire documentation of the "tls" stanza seems to be "settings for negotiating communication with the CA; includes acceptable ciphersuites, min/max TLS version, etc."

... "etc."? What fields are there? What are their valid values? What are their meanings?

I mean, I know that there's an example given on the same page, but that example has a field which is not mentioned (at least not explicitly or clearly) - "renegotiation", which to me implies maybe there are other unmentioned fields too, and I imagine there are more possible cipher suites than the two given in that example. Again, I just mean this as an example -- the whole document is largely this way, not just the "tls" portion.

**Comment by hslatman** [comment, marked as answer]
Hey @rwv37, this provides some more details based on the code, and should thus be more up-to-date: https://pkg.go.dev/github.com/smallstep/certificates@v0.26.1/authority/config. It's still not all encompassing, nor perfectly described, but at least from there you can explore the code more easily.
